### PR TITLE
add es torch effect test

### PR DESCRIPTION
**Pull request by rical730**
1. test_loss < 0.05
2. train_loss < 0.05
3. |test_loss - train_loss| < 0.03

### CODE REVIEW

File: deepes/test/torch_agent/torch_agent_test.cc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+#include <random>
+#include <math.h>
+
+using namespace DeepES;
+
+float evaluate(float* x_list, float* y_list, int size, std::shared_ptr<ESAgent<Model>> agent) {
+	float total_loss = 0.0;
+	for (int i = 0; i < size; ++i) {
+    // LOG(INFO) << "[x, y]" << x_list[i] << ", "<< y_list[i];
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/torch_agent/torch_agent_test.cc]: Please remove the log for debug

File: deepes/test/torch_agent/torch_agent_test.cc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	float* test_x_list = new float [test_data_size];
+	float* test_y_list = new float [test_data_size];
+
+	std::default_random_engine generator(0); // fix seed
+	std::uniform_real_distribution<float> uniform(-3.0, 9.0);
+	std::normal_distribution<float> norm;
+	for (int i = 0; i < train_data_size; ++i) {
+		x_list[i] = uniform(generator); // generate data between [0, 3]
+		y_list[i] = sin(x_list[i]) + norm(generator)*0.05; // noise std 0.05
+    // std::cout << " [x] " << x_list[i] <<  " [y] " << y_list[i] << std::endl;
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/torch_agent/torch_agent_test.cc]: remove the log here.

File: deepes/test/torch_agent/torch_demo_model.h [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    _obs_dim = obs_dim;
+    _act_dim = act_dim;
+    int hid1_size = act_dim * 5;
+    fc1 = register_module("fc1", torch::nn::Linear(obs_dim, hid1_size));
+    fc2 = register_module("fc2", torch::nn::Linear(hid1_size, act_dim));
+  }
+
+  torch::Tensor forward(torch::Tensor x) {
+    x = x.reshape({-1, _obs_dim});
+    // x = torch::tanh(fc1->forward(x));
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/torch_agent/torch_demo_model.h]: remove the log.

File: deepes/test/torch_agent/torch_agent_test.cc [hunk 4]
```diff
@@ -0,0 +1,111 @@
+//   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/torch_agent/torch_agent_test.cc]: Can we test the performance using the gtest framework ?

File: deepes/test/torch_agent/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required (VERSION 2.6)
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/torch_agent/CMakeLists.txt]: Why do we have another CmakeList.txt here? Can we use the prior CmakeLists.txt?

File: deepes/test/torch_agent/torch_agent_test.cc (hunk 1 above)
Comment (rical730) [reply deepes/test/torch_agent/torch_agent_test.cc]: done

File: deepes/test/torch_agent/torch_agent_test.cc (hunk 2 above)
Comment (rical730) [reply deepes/test/torch_agent/torch_agent_test.cc]: removed

File: deepes/test/torch_agent/torch_demo_model.h (hunk 3 above)
Comment (rical730) [reply deepes/test/torch_agent/torch_demo_model.h]: removed

File: deepes/test/torch_agent/torch_agent_test.cc (hunk 4 above)
Comment (rical730) [reply deepes/test/torch_agent/torch_agent_test.cc]: done

File: deepes/test/torch_agent/CMakeLists.txt (hunk 5 above)
Comment (rical730) [reply deepes/test/torch_agent/CMakeLists.txt]: done

File: deepes/test/src/torch_agent_test.cc [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+            return test_lo - train_lo;
+        }
+    }
+
+    float train_agent() {
+    }
+
+
+    void SetUp() override {
+        x_list = new float [train_data_size];
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/src/torch_agent_test.cc]: Please use vector<float> for data storage. It's more readable.

File: deepes/test/include/torch_demo_model.h [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+    _obs_dim = obs_dim;
+    _act_dim = act_dim;
+    int hid1_size = act_dim * 5;
+    fc1 = register_module("fc1", torch::nn::Linear(obs_dim, hid1_size));
+    fc2 = register_module("fc2", torch::nn::Linear(hid1_size, act_dim));
+  }
+
+  torch::Tensor forward(torch::Tensor x) {
+    x = x.reshape({-1, _obs_dim});
+    // x = torch::tanh(fc1->forward(x));
```
Comment (TomorrowIsAnOtherDay) [inline deepes/test/include/torch_demo_model.h]: remove the commented line.

Comment (rical730) [reply deepes/test/include/torch_demo_model.h]: removed

File: deepes/test/src/torch_agent_test.cc (hunk 6 above)
Comment (rical730) [reply deepes/test/src/torch_agent_test.cc]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]
Looks good to me.

**Pull request merged**